Escape input with html.escape, fixes HTTP 400

frebib opened the pull request:
Fixes responses like this, that contain HTML-like sequences:
{"ok":false,"error_code":400,"description":"Bad Request: can't parse entities: Unsupported start tag \"user@email.com\" at byte offset 860"}

Also split code over multiple lines for readability

Signed-off-by: Joe Groocock <me@frebib.net>

Adam-Ant commented:
Double check that inputs like isPR are escaped upstream to that field. Has it been confirmed that escaping here doesn't "double escape" when its passed into requests.json? 
Also, can we just import the escape function instead of the whole html library?

frebib committed:
Escape input with html.escape, fixes HTTP 400

Fixes responses like this, that contain HTML-like sequences:
{"ok":false,"error_code":400,"description":"Bad Request: can't parse entities: Unsupported start tag \"user@email.com\" at byte offset 860"}

Also split code over multiple lines for readability

Signed-off-by: Joe Groocock <me@frebib.net>

frebib commented:
Fixed your comments. I don't see any double escaping happening, this only escapes things like `<user@email.com>` which aren't supposed to be HTML in the first place

Adam-Ant commented:
LGTM
